Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -0,0 +1,148 @@
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-BqJob -ProjectId "my_project"</code>
+    ///   <para>Lists list all past or present jobs from the specified project, "my_project".</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> $job = Get-BqJob "job_p6focacVVo29rJ4_yvn8Aabi2wQ"</code>
+    ///   <para>This returns a descriptor object for the specified job in the default project.</para>
+    /// </example> 
+    /// <example>
+    ///   <code>PS C:\> $job = $job | Get-BqJob</code>
+    ///   <para>This will update the local descriptor $job with the most recent server state.</para>
+    /// </example> 
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs)">
+    /// [BigQuery Jobs]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "BqJob")]
+    public class GetBqJob : BqCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for jobs in. If not set via PowerShell parameter processing, it will
+        /// default to the Cloud SDK's default project.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public override string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the Job to get a reference for.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, Position = 0)]
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        public string JobId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Job object to get an updated reference for.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(JobList.JobsData),
+            Property = nameof(JobList.JobsData.JobReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Apis.Bigquery.v2.Data.Job),
+            Property = nameof(Apis.Bigquery.v2.Data.Job.JobReference))]
+        public JobReference InputObject { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            if (JobId == null && InputObject == null)
+            {
+                DoListRequest(Service.Jobs.List(Project));
+            }
+            else
+            {
+                if (JobId == null)
+                {
+                    JobId = InputObject.JobId;
+                }
+                DoGetRequest(Service.Jobs.Get(Project, JobId));
+            }
+        }
+
+        /// <summary>
+        /// Executes a List Jobs request and writes returned objects or errors.
+        /// </summary>
+        /// <param name="request">Pre-built request ready to be executed</param>
+        public void DoListRequest(JobsResource.ListRequest request)
```

Comment:
There appears to be a stateFilter in the Jobs.List API. As it is optional, it is probably a property rather than a parameter. There should be a Cmdlet parameter that sets it.
There is also an allUsers boolean. There should be a switch parameter for that as well.